google-cloud-compute/src/main/java/com/google/cloud/compute/nativeimage/ComputeFeature.java

```diff
@@ -0,0 +1,3657 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.compute.nativeimage;
+
+import com.google.api.gax.nativeimage.NativeImageUtils;
+import com.google.common.collect.ImmutableList;
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import org.graalvm.nativeimage.hosted.Feature;
+
+@AutomaticFeature
+class ComputeFeature implements Feature {
+
+  private static final ImmutableList<String> protoGoogleCommonProtosClasses =
+      ImmutableList.of(
+          "com.google.protobuf.DescriptorProtos$FieldOptions$Builder",
+          "com.google.protobuf.DescriptorProtos$FieldOptions$CType",
+          "com.google.protobuf.DescriptorProtos$FieldOptions$JSType",
+          "com.google.protobuf.DescriptorProtos$FieldOptions",
+          "com.google.protobuf.DescriptorProtos$MethodOptions$Builder",
+          "com.google.protobuf.DescriptorProtos$MethodOptions$IdempotencyLevel",
+          "com.google.protobuf.DescriptorProtos$MethodOptions",
+          "com.google.protobuf.DescriptorProtos$ServiceOptions$Builder",
+          "com.google.protobuf.DescriptorProtos$ServiceOptions",
+          "com.google.protobuf.DescriptorProtos$UninterpretedOption",
+          "com.google.api.FieldBehavior",
+          "com.google.cloud.OperationResponseMapping");
+
+  private static final ImmutableList<String> computeClasses;
+
+  static {
+    // Registering all GeneratedMessageV3 and builder classes in the proto directory; otherwise
+    // you'll get NoSuchMethodException at native image runtime. Here is how to generate the list:
+    // https://github.com/googleapis/java-compute/issues/685
+    ImmutableList.Builder<String> computeClassesBuilder = ImmutableList.builder();
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesInstanceGroupManagerRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesInstanceGroupManagerRequest$Builder");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesRegionInstanceGroupManagerRequest");
+    computeClassesBuilder.add(
+        "com.google.cloud.compute.v1.AbandonInstancesRegionInstanceGroupManagerRequest$Builder");
```

Comment:
Could you point me to the tests that fail if we don't include these `com.google.cloud.compute.v1` classes? I've experienced situations where I had to add a ton a classes for reflection to make the test work but it was just one class that was needed in the end, so I'm secretly hoping that we'll run into same scenario here.

Reply:
https://source.cloud.google.com/results/invocations/81073aee-a5fa-4581-af46-462972101862/targets/cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-compute%2Fpresubmit%2Fgraalvm-native/log

Here is an example:

```
  JUnit Vintage:NetworkFirewallPoliciesClientTest:getTest
    MethodSource [className = 'com.google.cloud.compute.v1.NetworkFirewallPoliciesClientTest', methodName = 'getTest', methodParameterTypes = '']
    => com.google.api.gax.rpc.UnknownException: java.lang.RuntimeException: Generated message class "com.google.cloud.compute.v1.FirewallPolicy" missing method "getAssociationsList".
       com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:119)
       com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:41)
       com.google.api.gax.httpjson.HttpJsonApiExceptionFactory.create(HttpJsonApiExceptionFactory.java:67)
       com.google.api.gax.httpjson.HttpJsonExceptionCallable$ExceptionTransformingFuture.onFailure(HttpJsonExceptionCallable.java:97)
       com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
       [...]
       Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
         com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
         com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClient.get(NetworkFirewallPoliciesClient.java:687)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClient.get(NetworkFirewallPoliciesClient.java:662)
         com.google.cloud.compute.v1.NetworkFirewallPoliciesClientTest.getTest(NetworkFirewallPoliciesClientTest.java:388)
         [...]
     Caused by: java.lang.RuntimeException: Generated message class "com.google.cloud.compute.v1.FirewallPolicy" missing method "getAssociationsList".
       com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1847)
       com.google.protobuf.GeneratedMessageV3.access$1000(GeneratedMessageV3.java:79)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedFieldAccessor$ReflectionInvoker.<init>(GeneratedMessageV3.java:2409)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedFieldAccessor.<init>(GeneratedMessageV3.java:2479)
       com.google.protobuf.GeneratedMessageV3$FieldAccessorTable$RepeatedMessageFieldAccessor.<init>(GeneratedMessageV3.java:3004)
       [...]
     Caused by: java.lang.NoSuchMethodException: com.google.cloud.compute.v1.FirewallPolicy.getAssociationsList()
       java.lang.Class.getMethod(DynamicHub.java:1114)
       com.google.protobuf.GeneratedMessageV3.getMethodOrDie(GeneratedMessageV3.java:1844)
       [...]
```